src/TaskCreationSection/CategoryCreationSection/IconPicker/IconPicker.jsx

```diff
@@ -1,61 +1,65 @@
-/* eslint-disable no-use-before-define */
-/* eslint-disable import/prefer-default-export */
+/* eslint-disable react/no-unstable-nested-components */
 /* eslint-disable react/jsx-no-constructed-context-values */
 /* eslint-disable react/prop-types */
-import React, { useMemo, useState, useRef } from "react";
+import React, { useMemo } from "react";
 import { IconContext } from "react-icons";
 import * as icons from "react-icons/io";
 import { colorStyleInputHandler } from "../../../colorStyleClassHandler";
 import DynamicIcon from "./DynamicIcon";
 
-const toArray = (z) => Object.values(z);
-const getLimitIcons = (z, limit = 25) => toArray(z).slice(0, limit);
-const findIcons = (z, searchIcon) =>
-  icons.filter((icon) =>
-    icon.name.toLowerCase().includes(searchIcon.toLowerCase())
-  );
-const debounceTime = 300;
-
-export function IconPicker({
+function IconPicker({
   newCategoryIcon,
   setNewCategoryIcon,
   colorStyle,
+  searchIcon,
+  setSearchIcon,
 }) {
-  const [searchIcon, setSearchIcon] = useState("");
-  const [iconsToDisplay, setIconsToDisplay] = useState(toArray(icons));
-  const loadingIcons = useRef(false);
   const onInput = (e) => {
     setSearchIcon(e.target.value);
-    handleLoadingIcons(e.target.value);
-  };
-
-  const handleLoadingIcons = (value) => {
-    if (!value) {
-      setIconsToDisplay(toArray(icons));
-    } else if (!loadingIcons.current) {
-      loadingIcons.current = true;
-      setTimeout(() => {
-        setIconsToDisplay(findIcons(getLimitIcons(icons), searchIcon));
-        loadingIcons.current = false;
-      }, debounceTime);
-    }
   };
+  const memoizedResult = useMemo(() => {
```

Comment:
readability: try to use variables as place where you can share more information/context with next developer that will review this code

so instead of `memoizedResult` you can use `filteredIcons` or `memoizedFilteredIcons` If you want to explicitly say that this result is memoized